crates/full-node/sov-db/src/storage_manager/nomt_based/groups.rs

```diff
@@ -62,6 +77,8 @@ where
     }
 
     pub(crate) fn commit(&mut self, group: CommitGroup) -> anyhow::Result<()> {
+        // The last commit had to be successful.
+        debug_assert_eq!(&self.commit_flag.read_status()?, &CommitStatus::Success);
```

Comment:
Non blocking: What if it is going to be normal assert? It will add couple ms to the commit time, which is not great, but give extra safety